scripts/inventory.js

```diff
@@ -184,34 +191,31 @@ async function loadInventoryData() {
  * @throws {Error} If the fetch request fails or data processing fails.
  */
 async function createCommunityInventoryMap() {
-  const response = await fetch('/data/hubblehomes.json?sheet=inventory');
+  // Load inventory data using the loadInventoryData function
+  const inventoryData = await loadInventoryData();
+  
   const models = await getModels();
 
-  if (response.ok) {
-    const inventory = await response.json();
-
-    // load the inventory and create a map of communities to homes
-    const communityMap = new Map();
+  // Create a map of communities to homes
+  const communityMap = new Map();
 
-    inventory.data.forEach((inventoryHome) => {
-      // inject the model image into the inventory home
-      const { image } = models.find((model) => model['model name'] === inventoryHome['model name']);
-      if (image) {
-        inventoryHome.image = image;
-      }
+  inventoryData.forEach((inventoryHome) => {
+    // Inject the model image into the inventory home
+    const { image } = models.find((model) => model['model name'] === inventoryHome['model name']) || {};
+    if (image) {
+      inventoryHome.image = image;
+    }
 
-      const { community } = inventoryHome;
-      if (!communityMap.has(community)) {
-        communityMap.set(community, []);
-      }
-      communityMap.get(community)
-        .push(inventoryHome);
-    });
+    const { community } = inventoryHome;
+    if (!communityMap.has(community)) {
+      communityMap.set(community, []);
+    }
+    communityMap.get(community).push(inventoryHome);
+  });
 
-    hh.inventory = communityMap;
-    return hh.inventory;
-  }
-  throw new Error('Failed to load inventory data');
+  // Cache the community map
+  hh.communityInventory = communityMap;
```

Comment:
I didn't even consider throwing this into the cache, I guess we could, but it would duplicating the same data twice.  The the method could just iterate and return the map.

Reply:
In the subsequent commits, I removed the logic to put the map in the cache, the function will create and return the map on invocation.